op-batcher/batcher/driver.go

```diff
@@ -28,6 +29,9 @@ type BatchSubmitter struct {
 	txMgr txmgr.TxManager
 	wg    sync.WaitGroup
 
+	// publishReceiveMu solve race condition: publishTxToL1 and handleReceipt access state concurrently
+	publishReceiveMu sync.Mutex
```

Comment:
Can we push this lock down into the `channelManager` and have it be over the public methods & document that the `channelManger` is safe for concurrent use? That would be a little less ad-hoc than this approach.

We're happy to let you take it on our take it on ourselves if you'd like.